mlir/include/mlir/Analysis/Presburger/PresburgerSpace.h

```diff
@@ -173,6 +173,7 @@ class PresburgerSpace {
     return PresburgerSpace(/*numDomain=*/0, /*numRange=*/numDims, numSymbols,
                            numLocals);
   }
+  PresburgerSpace() = default;
```

Comment:
There shouldn't be a default constructor for PresburgerSpace. PresburgerSpace has members which need to be initialized to maintain invariants (Identifiers). Please remove the default constructor.

Reply:
Well, I need the default constructor when I initialize ParserImpl. Again, either the existing ParseInfo was fine, or you let me uglify PresburgerSpace for the purposes of the parser.